apiserver/application/application.go

```diff
@@ -57,6 +60,15 @@ type API struct {
 	// should pass a charm.Charm and charm.URL back into
 	// state wherever we pass in a state.Charm currently.
 	stateCharm func(Charm) *state.Charm
+
+	deployApplicationFunc func(backend Backend, args jjj.DeployApplicationParams) error
+}
+
+// DeployApplication is a wrapper around juju.DeployApplication, to
+// match the function signature expected by NewAPI.
+func DeployApplication(backend Backend, args jjj.DeployApplicationParams) error {
```

Comment:
And I'd rather we didn't have hard-coded dependencies. Exporting this is a step towards being able to pass the dependency in, but providing an easy to get to default.